controller/src/main/java/org/jboss/as/controller/operations/global/ReadAttributeHandler.java

```diff
@@ -284,6 +284,16 @@ static void resolveAttribute(OperationContext context, AttributeDefinition attri
             // No defined value in the model. See if we should reply with a default from the metadata,
             // reply with undefined, or fail because it's a non-existent attribute name
             context.getResult().set(attribute.getDefaultValue());
+            // Reply with undefined when the attribute has declared alternatives and one of those alternatives is defined.
+            if (attribute.getAlternatives() != null) {
+                final ModelNode undefined = new ModelNode();
+                for(String alternative : attribute.getAlternatives()) {
+                    if(!undefined.equals(subModel.get(alternative))) {
+                        context.getResult().set(undefined);
+                        break;
+                    }
+                }
+            }
```

Comment:
I fixed the code with using hasAlternative method. I found and applied hasAlternative method of AttributeDefinition object to check alternatives as a callout to a separate method.  Please review the change. :-)